lib/topologies/server.js

```diff
@@ -247,8 +249,11 @@ var eventHandler = function(self, event) {
       // Query options
       var queryOptions = { numberToSkip: 0, numberToReturn: -1, checkKeys: false, slaveOk: true };
       // Create a query instance
-      var query = new Query(self.s.bson, 'admin.$cmd', {ismaster:true, client: self.clientInfo}, queryOptions);
-      // Get start time
+      if (self.s.compression && self.s.compression.compressors) {
```

Comment:
Thanks for that more simple version – I've implemented it now.

We accept a `compression` option (with properties `compressors` and `zlibCompressionLevel`) from the user, but we then need to flatten it down to just an array of compressors for the handshake with the server (according to the spec). Not sure if this answers your question…?